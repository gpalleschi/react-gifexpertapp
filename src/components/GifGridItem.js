import React from 'react'
import PropTypes from 'prop-types'; // IMPORT


export const GifGridItem = ( {url, title} ) => {
  return (
    <div className="card animate__animated animate__fadein">
         <img src={url} alt={title}></img>
	 <p>{ title }</p>
    </div>
  )
}

// SOTTO IL PROP TYPES :
GifGridItem.propTypes = {

url : PropTypes.string.isRequired,
title : PropTypes.string.isRequired
}

export default GifGridItem;



