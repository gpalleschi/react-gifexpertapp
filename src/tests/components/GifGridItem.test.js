import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />>', () => {
        const url = 'https://media1.giphy.com/media/yo3TC0yeHd53G/giphy.gif?cid=11889665o4rvrqhjl7ij3lcdwo6p7mpw3d3brn24hgdxmkv4&rid=giphy.gif&ct=g';
        const title = `saitama GIF`;
        const wrapper = shallow( <GifGridItem id={ 1 } url={ url } title={ title } /> );

	test('debe de mostrar <GifGridItem /> correctamente', () => {
            expect( wrapper ).toMatchSnapshot();
        });

	test('debe de tener un parrafo con el title', () => {
            const p = wrapper.find('p');
	    expect(p.text().trim()).toBe( title);		
	});

	test('debe de tener la imagen igual a la url y alt de los props ', () => {
		const img = wrapper.find('img');

		expect(img.prop('src')).toBe(url);
		expect(img.prop('alt')).toBe(title);

	});

	test('debe de tener la classe animate__fadein ', () => {

		const div = wrapper.find('div');

		expect(div.prop('className').includes('animate__fadein')).toBe(true);

	});	
});

