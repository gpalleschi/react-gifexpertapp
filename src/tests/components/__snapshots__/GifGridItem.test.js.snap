// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem />> debe de mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadein"
>
  <img
    alt="saitama GIF"
    src="https://media1.giphy.com/media/yo3TC0yeHd53G/giphy.gif?cid=11889665o4rvrqhjl7ij3lcdwo6p7mpw3d3brn24hgdxmkv4&rid=giphy.gif&ct=g"
  />
  <p>
    saitama GIF
  </p>
</div>
`;
